dds/DCPS/DataReaderImpl_T.h

```diff
@@ -250,9 +260,10 @@ namespace OpenDDS {
           ptr->instance_state_->sample_info(sample_info, item);
           item->sample_state_ = DDS::READ_SAMPLE_STATE;
 
-          if (observer) {
-            Observer::Sample s(*item, sample_info.instance_handle, sample_info.instance_state);
-            observer->on_sample_read(this, s);
+          const ValueWriterDispatcher* vwd = get_value_writer_dispatcher();
+          if (observer && item->registered_data_ && vwd) {
+            Observer::Sample s(sample_info.instance_handle, sample_info.instance_state, item->source_timestamp_, item->sequence_, item->registered_data_);
```

Comment:
should there be a constructor that takes a `ReceivedDataElement` and gets these 3 parts out of it?

Reply:
I could go either way so I'll go ahead an add one.